### ISSUE TITLE
Do not change MRU order when navigating while holding down modifier keys

### ISSUE THREAD

**Issue by rcaught** [Microsoft/vscode#17033]
When searching for the correct tab (ie. moving through tabs with `cmd+shift+]` and not letting go of `cmd+shift`), VS Code currently switches back to the last tab that it moved over when `ctrl+tab` is pressed.
![](http://g.recordit.co/88l5H2zhI6.gif)

I suggest that VS Code implements the behaviour of making the last used tab equal to that of the "last used tab".  Here is Sublime in action:
![](http://g.recordit.co/bonRoR2Vci.gif)

Sublime will always make the last used tab, that which was active before the key commands were held down for moving between tabs.

I understand that you could click on the tab with your mouse, find it in the quick search or use other keyboard commands, but I work with a lot more tabs than the contrived gifs suggest and I believe that Sublime has the correct behaviour.  It would be good to see this in VS Code.

**Comment by bpasero** [comment]
@rcaught I appreciate your detailed explanation but I am lost in understanding what you are asking for :-/. This is hard to understand:

> I suggest that VS Code implements the behaviour of making the last used tab equal to that of the "last used tab".

Can you maybe write down an example how you expect it to work with some steps?

**Comment by bpasero** [comment]
I am also having difficulties to find out which command is bound to `cmd+shift+]` in VS Code and Sublime (I am on a german keyboard layout).

**Comment by rcaught** [comment]
Ah sorry about that, I'm using the Sublime keymap for VS Code.  Everywhere that I wrote `cmd+shift+]` above should be `alt+cmd+arrow-right` (Next Editor).

Let me try again.  The feature is hard to explain (I nearly didn't write this issue because of that fact) but I'll try to outline it better with a written example.

- I'm on `tab 2` (but it could be any tab).  It is the tab I've been working in.
- I hold down `alt+cmd`, and I keep those two keys held down.  I now press either the right-arrow key or the left-arrow key to move through the tabs.
- Once I have found the tab that I want (let's say `tab 12` for this example), I let go of all the keys.
- I now press `ctrl+tab` (Next used editor in group).  This should open `tab 2` as `tab 2` was the last tab that I was actually using.

The current behaviour of VS Code would be to open `tab 11` or `tab 13` depending the direction I navigated there.

**Comment by bpasero** [comment]
@rcaught thanks this is understandable for me now. I see how this behaviour is desired, though I fear not as straight forward for us to implement because we do not distinguish navigating tabs via keybindings like that or actually clicking a tab by mouse.

We could probably solve this issue by opening our quick open list of tabs in the same way as we do it when you press and hold `Cmd+Tab`, but that is a bit ugly and might not be expected.

Let me update the issue title to make it easier to understand (at least for me).
